sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -364,7 +364,7 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case e @ EvaluatePython(udf, child, _) =>
         BatchPythonEvaluation(udf, e.output, planLater(child)) :: Nil
       case LogicalRDD(output, rdd) => PhysicalRDD(output, rdd, "ExistingRDD") :: Nil
-      case BroadcastHint(child) => apply(child)
+      case BroadcastHint(child) => planLater(child) :: Nil
```

Comment:
This is quite obvious, `apply` method will only make the rules within the object `BasicOperators` applied, but the data source(Parquet) is an extended strategy defined in another object, `planLater` means to search the all of the strategies(including the extensions), which is make more sense. That's why it caused exception says `No plan for BroadcastHint` as `apply(child)` returns `Nil`.